lab/src/Microsoft.ServiceModel.Syndication/src/Atom10FeedFormatter.cs

```diff
@@ -763,155 +868,160 @@ private SyndicationContent ReadContentFrom(XmlReader reader, SyndicationItem ite
-                                string name = reader.LocalName;
-                                if (FeedUtils.IsXmlns(name, ns) || FeedUtils.IsXmlSchemaType(name, ns))
+                                if (_preserveAttributeExtensions)
                                 {
-                                    continue;
-                                }
-                                string val = reader.Value;
-                                if (!TryParseAttribute(name, ns, val, result, this.Version))
-                                {
-                                    if (_preserveAttributeExtensions)
-                                    {
-                                        result.AttributeExtensions.Add(new XmlQualifiedName(reader.LocalName, reader.NamespaceURI), reader.Value);
-                                    }
-                                    else
-                                    {
-                                        TraceSyndicationElementIgnoredOnRead(reader);
-                                    }
+                                    result.AttributeExtensions.Add(new XmlQualifiedName(reader.LocalName, reader.NamespaceURI), await reader.GetValueAsync());
                                 }
                             }
                         }
                     }
-                    reader.ReadStartElement();
                 }
+                await reader.ReadStartElementAsync();
+            }
 
-                XmlBuffer buffer = null;
-                XmlDictionaryWriter extWriter = null;
-                bool areAllItemsRead = true;
-                bool readItemsAtLeastOnce = false;
-
-                if (!elementIsEmpty)
+            XmlBuffer buffer = null;
+            XmlDictionaryWriter extWriter = null;
+            bool areAllItemsRead = true;
+            bool readItemsAtLeastOnce = false;
+            
+            if (!elementIsEmpty)
+            {
+                try
                 {
-                    try
+                    while (await reader.IsStartElementAsync())
                     {
-                        while (reader.IsStartElement())
+                        if (await TryParseFeedElementFromAsync(reader, result))  
                         {
-                            if (TryParseFeedElementFrom(reader, result))
+                            // nothing, we parsed something, great
+                        }
+                        else if (await reader.IsStartElementAsync(Atom10Constants.EntryTag, Atom10Constants.Atom10Namespace) && !isSourceFeed)
+                        {
+                            if (readItemsAtLeastOnce)
                             {
-                                // nothing, we parsed something, great
+                                //throw new InvalidOperationException(String.Format(SR.FeedHasNonContiguousItems, this.GetType().ToString()));
+                                //Log we have disjoint items
```

Comment:
This is because the original parser crashed if there was disjoint items(entry). IE 
<title></title>
<entry></entry>
<entry></entry>
<date></date>
<entry></entry>

That would cause the program to fail. Jorge and I decided to refactor in order that if the feed is not well formatted, it wouldn't crash.

---
In reply to: [125739096](https://github.com/dotnet/wcf/pull/2028#discussion_r125739096) [](ancestors = 125739096)